### ISSUE TITLE
not identical on genesis blocks? 

### ISSUE THREAD

**Issue by immartian** [keorn/parity-spec#5]
I'm running some geth nodes and using a json file generated by this tool to run up a parity node. In theory they should be be able to sync to each other. However, I received this error after block #0: 

```
Stage 3 block verification failed for #1 (c421…aeed)
Error: Block(InvalidDifficulty(Mismatch { expected: 3806653, found: 3998047 }))
```
I noted Parity spec have more fields than geth one, but not so sure what exact reason they can't be talking to each other.

**Comment by keorn** [comment]
This tool has not been tested in all possible circumstances (it works with the example specs in the repo).
Seems like the difficulty ends up different. Could you please provide the chain spec for further debugging?

**Comment by immartian** [comment]
```
{"coinbase"   : "0x0000000000000000000000000000000000000000",
    "extraData"  : "",
        "nonce"      : "0x000000000000002a",
        "gasLimit"   : "0x7a1200",
    "difficulty" : "0x3d0900",
        "mixhash"    : "0x0000000000000000000000000000000000000000000000000000000000000000",
    "parentHash" : "0x0000000000000000000000000000000000000000000000000000000000000000",
    "timestamp"  : "0x00",
    "alloc": {
      "alloc": {
        "0000000000000000000000000000000000000001": {"balance": "1"},
        "0000000000000000000000000000000000000002": {"balance": "1"},
        "0000000000000000000000000000000000000003": {"balance": "1"},
        "0000000000000000000000000000000000000004": {"balance": "1"},
        "00756cf8159095948496617f5fb17ed95059f536": {"balance": "12345678901234567890123456789"},
                "9676ff525c8e0bdb41ac51ec304e2fd8107278fd": {"balance": "10000000000000000000000000"},
                "cA0214a1fc27489579D97401697925F253730881": {"balance": "100000000000000000000000000"},
        "d9b87b28449de9a45560fadace31300fcc50e68b": {"balance": "10000000000000000000000000"},
        "380271fa3742446636ab20a42c5ad2708a7572df": {"balance": "10000000000000000000000000"},
        "008d4c913ca41f1f8d73b43d8fa536da423f1fb4": {"balance": "10000000000000000000000000"},
        "575D25692f11dAAedfd4b1427C438Cc1687d54Cb": {"balance": "10000000000000000000000000"}
    }
    }
    }
```

with `--networkid= 55313716`  nothing else special

**Comment by immartian** [comment]
I have 2 enodes running  with this spec, but seems parity didn't match with error above:

enode://d383ed7364a61b49c8532cd96ed95bfaee49388779f04dd5baaf63769817595c6c2c16324d11598b4e90c1069e467a7aaf44f10ea20ae85da31752de60e0fc47@104.197.75.174:30303

enode://b5cd2f4e925f5eaa026e3965a89888e156d9e190f7d16cd685687c3ff0abbafe0a29a45d1ea2a0096f9454a07bdabf38f2fce3df108a33ce5620055016d7abac@104.198.70.194:30303

**Comment by immartian** [comment]
The issue persists even I changed some other specs, wonder what's the big diff between geth and parity. :

**Comment by keorn** [comment]
Have you tried the specs provided as an example?

**Comment by immartian** [comment]
I tried with same result, can you confirm the geth.json above tested as you wished?

**Comment by keorn** [comment]
I tried it a while ago and also had another person reporting success. At some point I might get around to figuring it out again, not much time now. You can also try starting with the Morden specs for both clients (which are known to work on the actual Morden) and modify them, let me know if you find something.

**Comment by immartian** [comment]
for the following result:

```
{
  "name": "Musicoin",
  "engine": {
    "Ethash": {
      "params": {
        "gasLimitBoundDivisor": "0x400",
        "minimumDifficulty": "0x20000",
        "difficultyBoundDivisor": "0x800",
        "durationLimit": "0xd",
        "blockReward": "0x4563918244F40000"
      }
    }
  },
  "params": {
    "accountStartNonce": "0x0",
    "maximumExtraDataSize": "0x20",
    "minGasLimit": "0x1388",
    "networkID": "0x34C0534"
  },
  "genesis": {
    "seal": {
      "ethereum": {
        "nonce": "0x000000000000002a",
        "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000"
      }
    },
    "difficulty": "0x3d0900",
    "author": "0x0000000000000000000000000000000000000000",
    "timestamp": "0x00",
    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "extraData": "0x",
    "gasLimit": "0x7a1200"
  },
  "nodes": [

  ],
  "accounts": {
    "0000000000000000000000000000000000000001": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "ecrecover",
        "pricing": {
          "linear": {
            "base": 3000,
            "word": 0
          }
        }
      }
    },
    "0000000000000000000000000000000000000002": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "sha256",
        "pricing": {
          "linear": {
            "base": 60,
            "word": 12
          }
        }
      }
    },
    "0000000000000000000000000000000000000003": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "ripemd160",
        "pricing": {
          "linear": {
            "base": 600,
            "word": 120
          }
        }
      }
    },
    "0000000000000000000000000000000000000004": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "identity",
        "pricing": {
          "linear": {
            "base": 15,
            "word": 3
          }
        }
      }
    },
    "00756cf8159095948496617f5fb17ed95059f536": {
      "balance": "12345678901234567890123456789",
      "nonce": "0x0"
    },
    "9676ff525c8e0bdb41ac51ec304e2fd8107278fd": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },
    "cA0214a1fc27489579D97401697925F253730881": {
      "balance": "100000000000000000000000000",
      "nonce": "0x0"
    },
    "d9b87b28449de9a45560fadace31300fcc50e68b": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },

    "380271fa3742446636ab20a42c5ad2708a7572df": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },
    "008d4c913ca41f1f8d73b43d8fa536da423f1fb4": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },
    "575D25692f11dAAedfd4b1427C438Cc1687d54Cb": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    }
  }
}
```

I can hardly see what's the problem with poor knowledge on Rust. The strange thing to me is block #0 seems synced, but stall at #1, what parameters may weigh here?

**Comment by keorn** [comment]
I am guessing the scenario is:
- start an instance of Geth and Parity
- connect them to each other
- mine block with Geth
- get the Error mentioned above

Could you provide more logs (all the way from startup) with `-l sync=trace`?

**Comment by immartian** [comment]
The error code found within this section:

```

2016-11-17 19:23:50  IO Worker #1 TRACE sync  New peer 0 (protocol: 63, network: 55313717, difficulty: Some(79812776), latest:a6f5…d22e, genesis:2e15…6c34, snapshot:None)
2016-11-17 19:23:50  IO Worker #1 DEBUG sync  Connected 0:0
2016-11-17 19:23:50  IO Worker #1 TRACE sync  Starting round (last imported count = None, block = 0
2016-11-17 19:23:50  IO Worker #1 TRACE sync  Starting sync with better chain
2016-11-17 19:23:50  IO Worker #1 TRACE sync  0 <- GetBlockHeaders: 256 entries starting from 2e15…6c34, set = NewBlocks
2016-11-17 19:23:51  IO Worker #3 TRACE sync  0 -> GetBlockHeaders (number: 1920000, max: 1, skip: 0, reverse:false)
2016-11-17 19:23:51  IO Worker #3 TRACE sync  0 -> GetBlockHeaders: returned 0 entries
2016-11-17 19:23:51  IO Worker #3 TRACE sync  0 -> BlockHeaders (1 entries), state = Blocks, set = NewBlocks
2016-11-17 19:23:51  IO Worker #3 TRACE sync  Header already in chain 0 (2e15…6c34), state = ChainHead
2016-11-17 19:23:51  IO Worker #3 TRACE sync  Received 1 subchain heads, proceeding to download
2016-11-17 19:23:51  IO Worker #3 TRACE sync  Imported 0 of 0
2016-11-17 19:23:51  IO Worker #3 TRACE sync  0 <- GetBlockHeaders: 128 entries starting from 2e15…6c34, set = NewBlocks
2016-11-17 19:23:51  IO Worker #3 TRACE sync  Syncing with 1/1 peers
2016-11-17 19:23:51  IO Worker #2 TRACE sync  == Connected 44: 44
2016-11-17 19:23:51  IO Worker #2 TRACE sync  Sending status to 44, protocol version 63
2016-11-17 19:23:51  IO Worker #1 TRACE sync  == Connected 31: 31
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Sending status to 31, protocol version 1
2016-11-17 19:23:51  IO Worker #2 TRACE sync  == Connected 55: 55
2016-11-17 19:23:51  IO Worker #2 TRACE sync  Sending status to 55, protocol version 1
2016-11-17 19:23:51  IO Worker #2 TRACE sync  == Connected 37: 37
2016-11-17 19:23:51  IO Worker #2 TRACE sync  Sending status to 37, protocol version 63
2016-11-17 19:23:51  IO Worker #1 TRACE sync  0 -> BlockHeaders (20 entries), state = Blocks, set = NewBlocks
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Header already in chain 0 (2e15…6c34)
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New head 2e15…6c34
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 2e1518061996c4edef23c4b6e41fc9688e4b58cca7388003ad06c859e6bc6c34
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 059aeabc559e6cb95b48d8d36bad1fa36d323fde09554c8cb2d0b8b0299996a4
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: aa987b424ef601ace981a862bf05ac4521b2fbd46a1f2f04e47ca612b9208794
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 24e34ce35f7a0d621602b38a05d3f58046c96020fd25d3da79bccbd60b1b1ca2
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 650aba9da0648ec3cf9d9c295fb7f61f00b024ec88ee950f83a609eba8e83987
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: d44ef889efa3f6e539bb06e60949667669a33beabb74024bcb6f17930a12dbe6
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 7aa70ae20910f34cf0c2b9b2f841a420bf92756424b91f63971eb5de2c89ffc2
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 31a090c67ca8151cdffb296823fda9d8ec00d189fe95124e854b9b406c900a24
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: a29fae5b84af0ddc33fb8277f21dc6a9df599538d9f3cbd96603f66b3f69e4f0
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 565c8dc56994369db13e183fe6c85742a94b39940cbfeeadc6af6e9642754870
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: bd07c62a6615a8c7572f376a0ddc7130205cb8a99c0ee1b6652546bda90dfcd5
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: b213d0b030ffcb017a66f50c7795094b001cdc937167a9b684c5b1ed7d7e62ec
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 07889a6b2afceae8ed025b6df74e8bca691c0379c5fb10f4bec99a0fed4e9c34
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: d6feef320472e57bdb3e71647593a42c15a37ea244eecbe8c8e93a575de15c58
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 6c5e547db67deac29647398404ff89ab157804d0013dc65e602252014dbb4e2a
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 62911896d9f0db257282df21273a76578c9cadf6421d81640ee2a59b141815db
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: b889af99dd705c37e1ec80a02a5c6565942dbb69ca29295c26e5272d3fff1616
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: 739b02814254be1e87bf822d231a69ab12defe3886283181a9852e4f9ac88366
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: d6ce48e32a603b9fc461f6a39d25ced39a2a076982bfd7461b3d685f17c468b4
2016-11-17 19:23:51  IO Worker #1 TRACE sync  New header: a6f5ef2fb933c183502a796845ff13745d7666ae9eddfcc48669dded1ee1d22e
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Inserted 20 headers
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Drained 20 blocks, new head :Some(a6f5ef2fb933c183502a796845ff13745d7666ae9eddfcc48669dded1ee1d22e)
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block already in chain 2e1518061996c4edef23c4b6e41fc9688e4b58cca7388003ad06c859e6bc6c34
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 059aeabc559e6cb95b48d8d36bad1fa36d323fde09554c8cb2d0b8b0299996a4
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued aa987b424ef601ace981a862bf05ac4521b2fbd46a1f2f04e47ca612b9208794
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 24e34ce35f7a0d621602b38a05d3f58046c96020fd25d3da79bccbd60b1b1ca2
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 650aba9da0648ec3cf9d9c295fb7f61f00b024ec88ee950f83a609eba8e83987
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued d44ef889efa3f6e539bb06e60949667669a33beabb74024bcb6f17930a12dbe6
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 7aa70ae20910f34cf0c2b9b2f841a420bf92756424b91f63971eb5de2c89ffc2
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 31a090c67ca8151cdffb296823fda9d8ec00d189fe95124e854b9b406c900a24
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued a29fae5b84af0ddc33fb8277f21dc6a9df599538d9f3cbd96603f66b3f69e4f0
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 565c8dc56994369db13e183fe6c85742a94b39940cbfeeadc6af6e9642754870
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued bd07c62a6615a8c7572f376a0ddc7130205cb8a99c0ee1b6652546bda90dfcd5
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued b213d0b030ffcb017a66f50c7795094b001cdc937167a9b684c5b1ed7d7e62ec
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 07889a6b2afceae8ed025b6df74e8bca691c0379c5fb10f4bec99a0fed4e9c34
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued d6feef320472e57bdb3e71647593a42c15a37ea244eecbe8c8e93a575de15c58
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 6c5e547db67deac29647398404ff89ab157804d0013dc65e602252014dbb4e2a
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 62911896d9f0db257282df21273a76578c9cadf6421d81640ee2a59b141815db
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued b889af99dd705c37e1ec80a02a5c6565942dbb69ca29295c26e5272d3fff1616
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued 739b02814254be1e87bf822d231a69ab12defe3886283181a9852e4f9ac88366
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued d6ce48e32a603b9fc461f6a39d25ced39a2a076982bfd7461b3d685f17c468b4
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Block queued a6f5ef2fb933c183502a796845ff13745d7666ae9eddfcc48669dded1ee1d22e
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Imported 19 of 20
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Sync round complete
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Syncing with 1/1 peers
2016-11-17 19:23:51  IO Worker #1 TRACE sync  Sync complete
**2016-11-17 19:23:52  Verifier #0 WARN client  Stage 3 block verification failed for #1 (059a…96a4)
Error: Block(InvalidDifficulty(Mismatch { expected: 3806653, found: 3998047 }))**
2016-11-17 19:23:52  IO Worker #3 TRACE sync  == Connected 46: 46
2016-11-17 19:23:52  IO Worker #3 TRACE sync  Sending status to 46, protocol version 63
2016-11-17 19:23:52  IO Worker #2 TRACE sync  == Disconnecting 44: 44

```

**Comment by immartian** [comment]
with this spec(parity):

```
{
  "name": "Musicoin",
  "engine": {
    "Ethash": {
      "params": {
        "gasLimitBoundDivisor": "0x400",
        "minimumDifficulty": "0x20000",
        "difficultyBoundDivisor": "0x800",
        "durationLimit": "0xd",
        "blockReward": "0x4563918244F40000"
      }
    }
  },
  "params": {
    "accountStartNonce": "0x0",
    "maximumExtraDataSize": "0x20",
    "minGasLimit": "0x1388",
    "networkID": "0x34C0535"
  },
  "genesis": {
    "seal": {
      "ethereum": {
        "nonce": "0x000000000000002a",
        "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000"
      }
    },
    "difficulty": "0x3d0900",
    "author": "0x0000000000000000000000000000000000000000",
    "timestamp": "0x00",
    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "extraData": "0x",
    "gasLimit": "0x7a1200"
  },
  "nodes": [
    "enode://35cfbbc5a3c9742a1d05be7934d6e7d4686b0115ffa09e9057056867d482e723283263f44292ad426194d6e9c862933372362d10f583e3b42e1843314d6552ef@104.198.101.250:30303"
      ],
  "accounts": {
    "0000000000000000000000000000000000000001": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "ecrecover",
        "pricing": {
          "linear": {
            "base": 3000,
            "word": 0
          }
        }
      }
    },
    "0000000000000000000000000000000000000002": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "sha256",
        "pricing": {
          "linear": {
            "base": 60,
            "word": 12
          }
        }
      }
    },
    "0000000000000000000000000000000000000003": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "ripemd160",
        "pricing": {
          "linear": {
            "base": 600,
            "word": 120
          }
        }
      }
    },
    "0000000000000000000000000000000000000004": {
      "balance": "1",
      "nonce": "0x0",
      "builtin": {
        "name": "identity",
        "pricing": {
          "linear": {
            "base": 15,
            "word": 3
          }
        }
      }
    },
    "00756cf8159095948496617f5fb17ed95059f536": {
      "balance": "12345678901234567890123456789",
      "nonce": "0x0"
    },
    "9676ff525c8e0bdb41ac51ec304e2fd8107278fd": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },
    "cA0214a1fc27489579D97401697925F253730881": {
      "balance": "100000000000000000000000000",
      "nonce": "0x0"
    },
    "d9b87b28449de9a45560fadace31300fcc50e68b": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },

    "380271fa3742446636ab20a42c5ad2708a7572df": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },
    "008d4c913ca41f1f8d73b43d8fa536da423f1fb4": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    },
    "575D25692f11dAAedfd4b1427C438Cc1687d54Cb": {
      "balance": "10000000000000000000000000",
      "nonce": "0x0"
    }
  }
}
```

**Comment by keorn** [comment]
I see multiple blocks queued, can you please start with a simpler setup when you start a new blockchain and have one block around? How you connect the nodes and how the block is mined?

**Comment by immartian** [comment]
2 blocks version with:

(remote) $> geth console      // get the enode info used by parity node below

(local) $> ./parity --chain musicoin.json -l sync=trace --no-discovery --bootnodes "enode://30f5e6dabc6713378972e057005a7299f7518b0c71bea432602bb1c26bd07737d1b1fb3f702615b8b30727337bdccea77a1af380d19fdc4b040f12a2858ea416@104.198.101.250:30303"

the log: 

```
Sending status to 0, protocol version 63
2016-11-18 17:35:53  IO Worker #3 TRACE sync  New peer 0 (protocol: 63, network: 55313717, difficulty: Some(7998047), latest:3a0f…937e, genesis:2e15…6c34, snapshot:None)
2016-11-18 17:35:53  IO Worker #3 DEBUG sync  Connected 0:0
2016-11-18 17:35:53  IO Worker #3 TRACE sync  Starting round (last imported count = None, block = 0
2016-11-18 17:35:53  IO Worker #3 TRACE sync  Starting sync with better chain
2016-11-18 17:35:53  IO Worker #3 TRACE sync  0 <- GetBlockHeaders: 256 entries starting from 2e15…6c34, set = NewBlocks
2016-11-18 17:35:54  IO Worker #0 TRACE sync  0 -> GetBlockHeaders (number: 1920000, max: 1, skip: 0, reverse:false)
2016-11-18 17:35:54  IO Worker #0 TRACE sync  0 -> GetBlockHeaders: returned 0 entries
2016-11-18 17:35:54  IO Worker #0 TRACE sync  0 -> BlockHeaders (1 entries), state = Blocks, set = NewBlocks
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Header already in chain 0 (2e15…6c34), state = ChainHead
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Received 1 subchain heads, proceeding to download
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Imported 0 of 0
2016-11-18 17:35:54  IO Worker #0 TRACE sync  0 <- GetBlockHeaders: 128 entries starting from 2e15…6c34, set = NewBlocks
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Syncing with 1/1 peers
2016-11-18 17:35:54  IO Worker #3 TRACE sync  == Connected 3: 3
2016-11-18 17:35:54  IO Worker #3 TRACE sync  Sending status to 3, protocol version 63
2016-11-18 17:35:54  IO Worker #0 TRACE sync  0 -> BlockHeaders (2 entries), state = Blocks, set = NewBlocks
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Header already in chain 0 (2e15…6c34)
2016-11-18 17:35:54  IO Worker #0 TRACE sync  New head 2e15…6c34
2016-11-18 17:35:54  IO Worker #0 TRACE sync  New header: 2e1518061996c4edef23c4b6e41fc9688e4b58cca7388003ad06c859e6bc6c34
2016-11-18 17:35:54  IO Worker #0 TRACE sync  New header: 3a0fc98904133f06b6318b0091d2bae95724e41ef68cd11ccbb94477743c937e
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Inserted 2 headers
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Drained 2 blocks, new head :Some(3a0fc98904133f06b6318b0091d2bae95724e41ef68cd11ccbb94477743c937e)
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Block already in chain 2e1518061996c4edef23c4b6e41fc9688e4b58cca7388003ad06c859e6bc6c34
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Block queued 3a0fc98904133f06b6318b0091d2bae95724e41ef68cd11ccbb94477743c937e
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Imported 1 of 2
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Sync round complete
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Syncing with 1/1 peers
2016-11-18 17:35:54  IO Worker #0 TRACE sync  Sync complete
2016-11-18 17:35:54  IO Worker #3 TRACE sync  == Connected 13: 13
2016-11-18 17:35:54  IO Worker #3 TRACE sync  Sending status to 13, protocol version 63
2016-11-18 17:35:54  IO Worker #3 TRACE sync  == Disconnecting 3: 3
2016-11-18 17:35:54  Verifier #0 WARN client  Stage 3 block verification failed for #1 (3a0f…937e)
Error: Block(InvalidDifficulty(Mismatch { expected: 3806653, found: 3998047 }))
2016-11-18 17:35:54  IO Worker #2 TRACE sync  == Disconnecting 13: 13
2016-11-18 17:35:55  IO Worker #3 TRACE sync  == Connected 12: 12
2016-11-18 17:35:55  IO Worker #3 TRACE sync  Sending status to 12, protocol version 1
2016-11-18 17:35:55  IO Worker #0 TRACE sync  == Connected 35: 35
2016-11-18 17:35:55  IO Worker #0 TRACE sync  Sending status to 35, protocol version 63
2016-11-18 17:35:55  IO Worker #3 TRACE sync  == Disconnecting 12: 12
```

**Comment by immartian** [comment]
maybe we can ask gitter guys to help?

**Comment by keorn** [comment]
Sure, you can have a look at the new Ropsten specs:
[Geth](https://dl.dropboxusercontent.com/u/4270001/testnet_genesis.json) and [Parity](https://raw.githubusercontent.com/ethcore/parity/master/ethcore/res/ethereum/ropsten.json)

I see Geth specs now have an additional field that should help with getting identical networks. Will add it to the translator.

**Comment by keorn** [comment]
Now the translator accounts for the additional fields and the Ropsten config is generated correctly.

**Comment by immartian** [comment]
cool, let me try again.

**Comment by immartian** [comment]
I'd say it's totally a mess system for Ethereum, the reason is on one breakpoint of hardforks, your new parameters can work tentatively, but how many times this will last?

**Comment by keorn** [comment]
Agreed, its a mess (help out if you can :). But most of the time it has to be dealt with only on the public networks.

**Comment by immartian** [comment]
thanks a lot, for this support. Will share gratitude later once our chain launched officially,  https://github.com/Musicoin
